app/controllers/submissions_controller.rb

```diff
@@ -0,0 +1,77 @@
+class SubmissionsController < ApplicationController
+  before_action :authenticate_user!, only: [:create, :create_comment]
+  before_action :authenticate_to_submit, only: :new
+
+  helper_method :posts, :post, :new_post, :new_comment
+
+  def new
+  end
+
+  def show
+  end
+
+  def index
+  end
+
+  def create
+    @post = Submission.new(post_params)
+    @post.user_id = current_user.id
+    if @post.save
+      flash[:alert] = "Your submission was saved"
+      redirect_to submission_path(@post.id)
+    else
+      flash[:alert] = @post.errors.messages
+      redirect_to new_submission_path
```

Comment:
I'd rather see `render :new` used here so that the data the user submitted is persisted on the form. As a poster, I'd hate to write a whole text blurb and have that lost because of a validation error.  The flash notification would need to be updated to use `flash.now` in order to continue to work.